src/components/pod/pod.style.ts

```diff
@@ -1,36 +1,51 @@
 import styled, { css } from "styled-components";
-import { margin } from "styled-system";
+import { margin, MarginProps } from "styled-system";
 
 import { baseTheme } from "../../style/themes";
 import Link from "../link";
 import { StyledContent as StyledLinkContent } from "../link/link.style";
 import IconButton from "../icon-button";
 import StyledIcon from "../icon/icon.style";
 
-const StyledPod = styled.div`
+export interface StyledPodProps extends MarginProps {
+  alignTitle: "left" | "center" | "right";
+  internalEditButton?: boolean;
+  height?: string;
+}
+const StyledPod = styled.div<StyledPodProps>`
+  ${margin}
   display: flex;
   align-items: flex-start;
   width: 100%;
   text-align: ${({ alignTitle }) => alignTitle};
   ${({ internalEditButton }) => internalEditButton && "position: relative"};
-  ${margin}
   ${({ height }) => height && `height: ${height}`};
 
   &:focus {
     outline: none;
   }
 `;
 
-const blockBackgrounds = (variant) =>
-  ({
-    primary: "var(--colorsUtilityYang100)",
-    secondary: "var(--colorsUtilityMajor025)",
-    tertiary: "var(--colorsUtilityMajor040)",
-    transparent: "var(--colorsUtilityMajorTransparent)",
-    tile: "var(--colorsUtilityYang100)",
-  }[variant]);
+const blockBackgrounds = {
+  primary: "var(--colorsUtilityYang100)",
+  secondary: "var(--colorsUtilityMajor025)",
+  tertiary: "var(--colorsUtilityMajor040)",
+  transparent: "var(--colorsUtilityMajorTransparent)",
+  tile: "var(--colorsUtilityYang100)",
+};
 
-const StyledBlock = styled.div`
+export interface StyledBlockProps {
```

Comment:
I'm happy for you to leave this as is if you prefer, but I notice that you're defining a lot of the types for the props multiple times. For example `variant` is defined in `StyledBlockProps`, `StyledFooterProps` and then in `PodProps` (admittedly it's optional here but it can be for the rest imo). It might be worth defining some shared props interfaces in the style file sharing it between the interfaces that need it and then exporting them so it can be used in the component file, at the very least will make maintaining easier.

```
interface SharedProps {
  variant?: "primary" | "secondary" | "tertiary" | "tile" | "transparent";
  ...
}

export interface StyledBlockProps extends SharedProps {
  ...
}

// component file
export interface PodProps extends StyledBlockProps, StyledFooterProps {
  ...
}
```